app/src/main/java/com/example/root/smsreader/MainActivity.java

```diff
@@ -22,117 +15,116 @@
 
 public class MainActivity extends AppCompatActivity {
 
-public  int len_all_sms = 0;
-public int Count=0;
+    private static final String TAG = MainActivity.class.getSimpleName();
+    // Member variables have prefix "m". rest of the name uses CamelCase
+public  int mLenAllSms = 0;
+public int mCount = 0;
 
 
-    public String INR[] ;
-    ListView L1 ;
-    TextView count;
-    Button analysis;
+    public String mINR[] ;
+    ListView mLvSms;
+    TextView mTvCount; // for TextView, add Tv after prefix of "m"
+    Button mBtnAnalysis;
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        
         setContentView(R.layout.activity_main);
         Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
         setSupportActionBar(toolbar);
-        Log.v("Test", "List view initialized");
-        List<String> s;
-        s = readSms();
-        len_all_sms = s.size();
-        Log.v("Test", "readSms func called");
-        Log.v("Test", " "+ len_all_sms +" objects will be created now .....");
+        Log.d(TAG, "List view initialized"); //use Log.d() . this is not run during in release builds. Logging affects performance
```

Comment:
Logging increases the amount of processing. It decreases the performance of the app. As the codebase grows, the number of Log statements can become quite a lot.
 **Using Log.d() ensures that the performance is not hit when the app is actually launched.**